src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptions.java

```diff
@@ -596,6 +597,98 @@ public boolean apply(Object input) {
     }
 
     /**
+     * Tests that emulates a frozen elected master node that unfreezes and pushes his cluster state to other nodes
+     * that already are following another elected master node. These nodes should reject this cluster state and prevent
+     * that they are going to follow the stale master.
+     */
+    @Test
+    public void testStaleMasterNotHijackingMajority() throws Exception {
+        // 3 node cluster with unicast discovery and minimum_master_nodes set to 2:
+        List<String> nodes = startUnicastCluster(3, null, 2);
```

Comment:
Due to the fact that on osx multicast threads are shared between nodes we can't suspend all the threads... is the same reason for the test above this one.